### ISSUE TITLE
StreamDeck disconnected?

### ISSUE THREAD

**Issue by philetaylor1** [OpenStreamDeck/StreamDeckSharp#3]
Hi.

I am writing a Windows Service and I have found that it would be useful if there was a way to test whether the StreamDeck was still connected before issuing commands to it, is this possible?

Thanks

Phil

**Comment by philetaylor1** [comment]
As a follow-up to this, I have successfully exposed the isConnected property from HIDLibrary and that allows me to check if the StreamDeck is connected. 

HIDLibrary also has Inserted/Removed events available so it might be nice to expose these as well?

Thanks

Phil

**Comment by wischi-chr** [comment]
At the moment streamdecksharp does not support plug&play. If the device is disconnected while writing an image it will stop working because there is no timeout on writing to the device.

Currently the image writing is a bit bodgy because it uses 15 threads. The library caches the writing requests for each key and if you write to a key multiple times very quickly it discards bitmaps in between to speed things up. 

Other libraries only allow syncronized communication (wait for HID response) but that's pretty slow.

Long story short. I need to do some refactoring before this feature can be added.

**Comment by wischi-chr** [comment]
StreamDeckSharp 0.1.4 with `ConnectionStateChanged` event is now available and disconnecting and reconnecting should work now. Please let me know if that works for you.
